components/gitpod-db/src/user-db.spec.db.ts

```diff
@@ -36,6 +36,7 @@ import { DBWorkspaceInstance } from './typeorm/entity/db-workspace-instance';
         const typeorm = testContainer.get<TypeORM>(TypeORM);
         const mnr = await typeorm.getConnection();
         await mnr.getRepository(DBUser).delete({});
+        await mnr.getRepository(DBIdentity).delete({});
```

Comment:
@geropl On a side note, while manipulating identities in `TeamDBSpec`, I stumbled on a weird failure in `UserDBSpec`:


```
  1) UserDBSpec
       createUserAndFindById:

      AssertionError: expected { Object (id, creationDate, ...) } to have deep property 'identities' of [ Array(1) ], but got [ Array(1) ]
      + expected - actual

           "authId": "1234"
           "authName": "gero"
           "authProviderId": "GitHub"
           "deleted": false
      -    "primaryEmail": "tom@example.com"
      +    "primaryEmail": [undefined]
           "readonly": false
           "tokens": []
         }
       ]
```

See: https://werft.gitpod-dev.com/job/gitpod-build-jx-optimize-get-team-members.2

That's when I realized that identities are **not** deleted when users are deleted:

https://github.com/gitpod-io/gitpod/blob/a6c548135d2ae3d1b10fb24cb1a07ae828ae4b55/components/gitpod-db/src/typeorm/entity/db-user.ts#L41-L51

My drive-by fix here was to also clean up the `DBIdentity` repo after cleaning up `DBUser` (in both `UserDBSpec` and `TeamDBSpec`). But that comment makes a good point -- maybe we should cascadeRemove identities?

Reply:
Wasn't the reason that we are not cascade deleting related items that it needs to go through dbsync? i.e. we need to enable soft-deletion for that.